service/src/main/java/bio/terra/catalog/common/DaoKeyHolder.java

```diff
@@ -0,0 +1,42 @@
+package bio.terra.catalog.common;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+
+public class DaoKeyHolder extends GeneratedKeyHolder {
+
+  public UUID getId() {
+    return getField("id", UUID.class).orElse(null);
+  }
+
+  public Timestamp getTimestamp(String fieldName) {
+    return getField(fieldName, Timestamp.class).orElse(null);
+  }
+
+  public Instant getCreatedDate() {
+    Timestamp timestamp = getTimestamp("created_date");
+    if (timestamp != null) {
+      return timestamp.toInstant();
+    }
+    return null;
+  }
+
+  public String getString(String fieldName) {
+    return getField(fieldName, String.class).orElse(null);
+  }
+
+  public <T> Optional<T> getField(String fieldName, Class<T> type) {
+    Map<String, Object> keys = getKeys();
+    if (keys != null) {
+      Object fieldObject = keys.get(fieldName);
+      if (type.isInstance(fieldObject)) {
```

Comment:
As far as I can tell, this is the main useful line in this method. Under what circumstances are we getting back an incorrectly typed object? In those cases do we want to ignore the type difference and return `null` instead? Unless we have a clear case where we know there's a type mismatch, I think it would be better to return `keys.get(fieldName)` cast to the expected type directly. A type mismatch will result in a thrown runtime exception, but I think we'd want to know about this and not hide the error.

Reply:
I borrowed this from the data repo, but that's a good point that we don't want to hide the error!